packages/esm-ward-app/src/location-selector/location-selector.component.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import styles from './location-selector.scss';
+import useLocations from '../hooks/useLocations';
+import { RadioButton, Search, RadioButtonGroup, RadioButtonSkeleton } from '@carbon/react';
+import { useTranslation } from 'react-i18next';
+import { Button } from '@carbon/react';
+import {
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  isDesktop,
+  ResponsiveWrapper,
+  useDebounce,
+  useLayoutType,
+} from '@openmrs/esm-framework';
+import classNames from 'classnames';
+import type { RadioButtonGroupProps } from '@carbon/react/lib/components/RadioButtonGroup/RadioButtonGroup';
+import useEmrConfiguration from '../hooks/useEmrConfiguration';
+
+const size = 5;
+
+interface LocationSelectorProps extends RadioButtonGroupProps {}
+
+export default function LocationSelector(props: LocationSelectorProps) {
+  const { t } = useTranslation();
+  const { emrConfiguration, isLoadingEmrConfiguration } = useEmrConfiguration();
+  const isTablet = !isDesktop(useLayoutType());
+  const [searchTerm, setSearchTerm] = useState('');
+  const debouncedSearchTerm = useDebounce(searchTerm);
+  const [page, setPage] = useState(1);
+  const filterCriteria: Array<Array<string>> = useMemo(() => {
+    const criteria = [];
+    if (debouncedSearchTerm) {
+      criteria.push(['name:contains', debouncedSearchTerm]);
+    }
+    criteria.push(['_count', size.toString()]);
+    if (emrConfiguration) {
+      criteria.push(['_tag', emrConfiguration.supportsTransferLocationTag.display]);
```

Comment:
Side note to @mseaton ... we should likely improve the EMR API configuration to return "name" for tags so we don't have to key on display.

Reply:
I believe I can do the same from the custom representation, let me try.
Thanks!

Reply:
Duh, you are right @vasharma05  and sorry to @mseaton , I had forgotten he build custom rep support for this, yes, that is what we should do here.